python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -194,14 +195,13 @@ def _get_runs_to_dequeue(
         in_progress_runs = self._get_in_progress_runs(instance)
 
         max_concurrent_runs_enabled = max_concurrent_runs != -1  # setting to -1 disables the limit
+        max_runs_to_launch = max_concurrent_runs - len_iter(in_progress_runs)
         if max_concurrent_runs_enabled:
-            max_runs_to_launch = max_concurrent_runs - len(in_progress_runs)
-
             # Possibly under 0 if runs were launched without queuing
             if max_runs_to_launch <= 0:
                 self._logger.info(
                     "{} runs are currently in progress. Maximum is {}, won't launch more.".format(
-                        len(in_progress_runs), max_concurrent_runs
+                        len_iter(in_progress_runs), max_concurrent_runs
```

Comment:
another thing I would address separately (maybe in a follow up). I'd love to have some agreed upon norms to handle `Mapping`, `Sequence`, and `Iterable` classes, because a lot of the idioms right now are annoying. For example this just came up yesterday https://github.com/dagster-io/dagster/pull/11713#discussion_r1070198129

Reply:
My approach is:

- Use `Mapping` everywhere for dict unless we intend to mutate the dict.
- Use `Sequence` everywhere for list unless we intend to mutate the list.
- `Iterable` doesn't come up frequently, I've assumed when it does appear it's because we have an abstract method that might return either a list or an iterator. It's possible there are some places in the code where `Iterable` should instead be sequence (like in this case, where we are checking length).